pkg/p2p/libp2p/internal/handshake/handshake.go

```diff
@@ -50,20 +47,33 @@ type Service struct {
 	network.Notifiee // handhsake service can be the receiver for network.Notify
 }
 
-func New(overlay swarm.Address, networkID uint64, logger logging.Logger) *Service {
+func New(overlay swarm.Address, underlay []byte, signer crypto.SignRecoverer, networkID uint64, logger logging.Logger) (*Service, error) {
```

Comment:
Hmm I agree, I would actually even use underlay as a string in proto messages as well, I don't see the reason for it to be a byte array. I will change to what u suggested either way

Reply:
Done. I left underlay as bytearray as per spec and the book. We can talk about it with Viktor and I can change it to `string` if accepted. It can go well with the Hive msg changes PR.